marketplaceItems/sitPoint/appResources/appData/sitClient.js

```diff
@@ -43,186 +43,152 @@
 
     // #endregion DEPENDENCIES
 
-    // #region SITTABLE OVERLAY - Overlay on chair "Click to Sit" or "Trigger to Sit"
-
-    // Constants
-    var OVERLAY_SITTABLE_FADE_MS = 50; // "Click/Trigger to Sit" overlay fade after 50 ms
-    var OVERLAY_SITTABLE_DISTANCE_SHOW_M = 3; // m 
-    var OVERLAY_SITTABLE_DISTANCE_MAX_M = 5; // m
-    var OVERLAY_SITTABLE_START_ALPHA = 0.7;
-    var OVERLAY_SITTABLE_MIN_END_ALPHA = 0.075; // fades to this alpha value
-    var OVERLAY_URL_SITTABLE_HMD = Script.resolvePath("./resources/images/triggerToSit.png");
-    var OVERLAY_URL_SITTABLE_DESKTOP = Script.resolvePath("./resources/images/clickToSit.png");
-
-    // Dynamic variables
-    var overlaySittable = null;
-    var intervalSittableTransparency = null;
-
-    // Create sittable overlay
+    // #region SITTABLE - Local entity image on chair "Click to Sit" or "Trigger to Sit"
+
+    // Create sittable local entity
+    var SITTABLE_FADE_MS = 50; // "Click/Trigger to Sit" local entity image fade after 50 ms
+    var SITTABLE_DISTANCE_SHOW_M = 3; // m 
+    var SITTABLE_DISTANCE_MAX_M = 5; // m
+    var SITTABLE_START_ALPHA = 0.7;
+    var SITTABLE_MIN_END_ALPHA = 0.075; // fades to this alpha value
+    var SITTABLE_IMAGE_URL_HMD = Script.resolvePath("./resources/images/triggerToSit.png");
+    var SITTABLE_IMAGE_URL_DESKTOP = Script.resolvePath("./resources/images/clickToSit.png");
+    var sittable_y_offset = 0.01; // default 0.01
     function sittableCreate() {
-
-        if (OVERLAY_DEBUG) {
+        if (UI_DEBUG) {
             print("SittableCreate");
         }
 
         setChairProperties();
 
         // change the image based on what modality the user is in
         var url = HMD.active
-            ? OVERLAY_URL_SITTABLE_HMD
-            : OVERLAY_URL_SITTABLE_DESKTOP;
+            ? SITTABLE_IMAGE_URL_HMD
+            : SITTABLE_IMAGE_URL_DESKTOP;
 
-        var overlayPosition = Vec3.sum(chairProperties.position, { x: 0, y: 0.01, z: 0 });
+        var localOffset = {
+            x: 0,
+            y: sittable_y_offset,
+            z: 0
+        };
+        var worldOffset = Vec3.multiplyQbyV(that.chairProperties.rotation, localOffset);
+        var sittablePosition = Vec3.sum(that.chairProperties.position, worldOffset);
 
-        overlaySittable = Overlays.addOverlay("image3d", {
-            position: overlayPosition,
-            rotation: Quat.multiply(chairProperties.rotation, Quat.fromVec3Degrees({ x: -90, y: 180, z: 0 })),
+        that.sittableID = Entities.addEntity({
```

Comment:
since `sittableCreate` isn't a member function of the prototype, `that` is referring to a global variable and not one passed directly to the function. you could solve that by making `sittableCreate` a member function.

Reply:
That is referencing the scoped "that" in the script. I don't think the createSittable function needs to be added to the prototype since the helper functions reference the "that" variable. 

Happy to jump on a call to discuss.